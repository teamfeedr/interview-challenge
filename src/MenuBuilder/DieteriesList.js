import React from 'react';

const DieteriesList = ({ dietaries = [] }) => (
  <p>
    {dietaries.map(diet => (
      <span key={diet} className="dietary">
        {diet}
      </span>
    ))}
  </p>
);

export default DieteriesList;
