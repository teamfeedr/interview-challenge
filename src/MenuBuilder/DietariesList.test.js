import React from 'react';
import DieteriesList from './DietariesList';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders empty', () => {
  const div = shallow(<DieteriesList />);
  expect(div).toMatchSnapshot();
});

it('renders a list of dietery types', () => {
  const dietaries = ['v', 'n!'];
  const div = shallow(<DieteriesList dietaries={dietaries} />);
  expect(div).toMatchSnapshot();
});
