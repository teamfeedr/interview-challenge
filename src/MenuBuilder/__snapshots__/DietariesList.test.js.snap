// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of dietery types 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DieteriesList
    dietaries={
        Array [
            "v",
            "n!",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="dietary"
>
          v
</span>,
        <span
          className="dietary"
>
          n!
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "v",
        "nodeType": "host",
        "props": Object {
          "children": "v",
          "className": "dietary",
        },
        "ref": null,
        "rendered": "v",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "n!",
        "nodeType": "host",
        "props": Object {
          "children": "n!",
          "className": "dietary",
        },
        "ref": null,
        "rendered": "n!",
        "type": "span",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="dietary"
>
            v
</span>,
          <span
            className="dietary"
>
            n!
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "v",
          "nodeType": "host",
          "props": Object {
            "children": "v",
            "className": "dietary",
          },
          "ref": null,
          "rendered": "v",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "n!",
          "nodeType": "host",
          "props": Object {
            "children": "n!",
            "className": "dietary",
          },
          "ref": null,
          "rendered": "n!",
          "type": "span",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DieteriesList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
